File: performance-comparison/benchmarks/data_structures/clojure/sort.clj
(let [n   (Long/parseLong (first *command-line-args*))
      arr (loop [i 0 s (long 12345) acc (transient [])]
            (if (>= i n) (persistent! acc)
                (let [ns (unchecked-add (unchecked-multiply s 6364136223846793005)
                                        1442695040888963407)]
                  (recur (inc i) ns (conj! acc (unsigned-bit-shift-right ns 1))))))
      sorted (sort arr)]
  (println (first sorted) (last sorted)))

File: performance-comparison/benchmarks/data_structures/python/sort.py
import sys

n = int(sys.argv[1]) if len(sys.argv) > 1 else 1000
mask = (1 << 64) - 1
s = 12345
arr = []
for _ in range(n):
    s = (s * 6364136223846793005 + 1442695040888963407) & mask
    arr.append(s >> 1)
arr.sort()
print(arr[0], arr[-1])
